### PR TITLE
Enabled tests to read M2_REPO env vars

### PR DESCRIPTION
**Pull request by desmax74**
@mbiarnes @jhrcek

### CODE REVIEW

**Comment by desmax74** [comment]
jenkins test

File: kie-wb-common-services/kie-wb-common-compiler/kie-wb-common-compiler-service/src/test/java/org/kie/workbench/common/services/backend/compiler/BaseCompilerTest.java [hunk 1]
```diff
@@ -63,6 +58,30 @@ public BaseCompilerTest(String prjName) {
...
+        return StringUtils.isEmpty(mavenRepo) ? createMavenRepo().toAbsolutePath().toString() : mavenRepo;
+    }
+
+    public static Path createMavenRepo() throws Exception {
+        Path mavenRepository = Paths.get(System.getProperty("user.home"),
+                                         "/.m2/repository");
+        if (!Files.exists(mavenRepository)) {
+            logger.info("Creating a m2_repo into " + mavenRepository);
+            if (!Files.exists(Files.createDirectories(mavenRepository))) {
+                throw new Exception("Folder not writable in the project");
```
Comment (jhrcek) [inline kie-wb-common-services/kie-wb-common-compiler/kie-wb-common-compiler-service/src/test/java/org/kie/workbench/common/services/backend/compiler/BaseCompilerTest.java]: Optional nitpick: you should probably make this exception message more informative and add info about which folder is not writeable to the exception message.

File: kie-wb-common-services/kie-wb-common-compiler/kie-wb-common-compiler-service/src/test/java/org/kie/workbench/common/services/backend/compiler/BaseCompilerTest.java [hunk 2]
```diff
@@ -63,6 +58,30 @@ public BaseCompilerTest(String prjName) {
         }
     }
 
+    public static String getMavenRepo() throws Exception {
+        List<String> repos = Arrays.asList("M2_REPO", "MAVEN_REPO_LOCAL", "MAVEN_REPO", "M2_REPO_LOCAL");
```
Comment (jhrcek) [inline kie-wb-common-services/kie-wb-common-compiler/kie-wb-common-compiler-service/src/test/java/org/kie/workbench/common/services/backend/compiler/BaseCompilerTest.java]: I don't like that the code is duplicated in 4 places. Can't this be consolidated into one? I realize that it's multiple maven modules but you could have one testing module with all the utilities and use it as test dependency in all theses maven modules..

Comment (desmax74) [reply kie-wb-common-services/kie-wb-common-compiler/kie-wb-common-compiler-service/src/test/java/org/kie/workbench/common/services/backend/compiler/BaseCompilerTest.java]: Fixed

Comment (desmax74) [reply kie-wb-common-services/kie-wb-common-compiler/kie-wb-common-compiler-service/src/test/java/org/kie/workbench/common/services/backend/compiler/BaseCompilerTest.java]: Ok, I'll fix this testing issue in the offprocess PR

Comment (desmax74) [reply kie-wb-common-services/kie-wb-common-compiler/kie-wb-common-compiler-service/src/test/java/org/kie/workbench/common/services/backend/compiler/BaseCompilerTest.java]: Fixed in a central module

**Comment by desmax74** [comment]
jenkins test

**Comment by desmax74** [comment]
@porcelli

**Comment by ederign** [comment]
@jhrcek this is ok from QE?

**Comment by desmax74** [comment]
jenkins test

**Comment by desmax74** [comment]
jenkins test

**Comment by desmax74** [comment]
@ederign @porcelli is green

**Pull request merged**